### PR TITLE
Disable buttons on employment and education dialogs during profile update

### PR DESCRIPTION
**Pull request by noisecapella**
#### What are the relevant tickets?
Part of #1752

#### What's this PR do?
Disables buttons for saving during profile upload for Education and Employment dialogs

#### How should this be manually tested?
Verify that the save buttons are disabled and that a blue spinner is shown in the buttons place. This should happen for employment and education, for both add and edit, and for both the signup page and the user page.

### CODE REVIEW

**Comment by aliceriot** [comment]
functionality looks great!

File: static/js/components/EducationForm.js [hunk 1]
```diff
@@ -161,7 +164,7 @@ class EducationForm extends ProfileFormFields {
       userPrivilegeCheck(profile, () =>
         <Cell col={12} className="profile-form-row add" key="add-row">
           <button
-            className="mm-minor-action"
+            className="mm-minor-action add-education-button"
```
Comment (aliceriot) [inline static/js/components/EducationForm.js]: what's the reason for this change?

File: static/js/containers/UserPage_test.js [hunk 2]
```diff
@@ -552,6 +552,29 @@ describe("UserPage", function() {
...
+          return listenForActions([
+            REQUEST_PATCH_USER_PROFILE,
+          ], () => {
+            helper.store.dispatch(requestPatchUserProfile(username));
+          }).then(() => {
+            let button = dialog.querySelector(".save-button");
+            assert(button.className.includes("disabled-with-spinner"));
+
+            TestUtils.Simulate.click(button);
+            assert.isFalse(patchUserProfileStub.called);
```
Comment (aliceriot) [inline static/js/containers/UserPage_test.js]: Maybe check that there's an element with class `.mdl-spinner` here too? Not sure how important that is.

File: static/js/components/EducationForm.js (hunk 1 above)
Comment (aliceriot) [inline static/js/components/EducationForm.js]: oh, I see it being used in the test below, is that the reason for it?

Comment (noisecapella) [inline static/js/components/EducationForm.js]: yes

File: static/js/containers/UserPage_test.js (hunk 2 above)
Comment (noisecapella) [inline static/js/containers/UserPage_test.js]: Sure, makes sense
